sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEShuffleReadExec.scala

```diff
@@ -183,6 +194,17 @@ case class AQEShuffleReadExec private(
         } else {
           Map.empty
         }
+      } ++ {
+        if (isLocalRead) {
+          Map.empty
+        } else {
+          if (hasCoalescedPartition) {
```

Comment:
reply to https://github.com/apache/spark/pull/32776#discussion_r676750494

If this is a local reader, then `hasCoalescedPartition` can never be true, right? I still think the `if (isLocalRead)` above is not needed.

Reply:
yes, you're right

Reply:
updated